### ISSUE TITLE
pytest yield tests

### ISSUE THREAD

**Issue by jantman** [jantman/awslimitchecker#233]
Pytest is now throwing a bunch of warnings about yield tests scheduled to be deprecated (see below).

Figure out what's involved in converting these to new-style tests, if that's even possible. If it is, do it. If it isn't, and there's no clear alternative, pin pytest to <4.0 and suppress the warning :(

```
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/test_integration.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/jantman/GIT/awslimitchecker/awslimitchecker/tests/services/test_base.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
```

**Comment by jantman** [comment]
Fixed in #237

**Comment by jantman** [comment]
This has been released in 0.7.0 and is live on pypi.
